python/imzml_reader.py

```diff
@@ -199,26 +214,34 @@ def write_metadata_json(
     )
     # FileType('x'): exclusive file creation, fails if file already exits.
     parser.add_argument(
-        "--ims_metadata",
+        "--raster_json",
         type=argparse.FileType("x"),
         required=True,
         help="Write the metadata about the IMS zarr store on S3.",
     )
     parser.add_argument(
-        "--zarr_store_url",
+        "--raster_name", required=True, help="Image name for metadata.",
+    )
+    parser.add_argument(
+        "--server_url",
         required=True,
         help="Destination for zarr output in cloud.",
     )
     args = parser.parse_args()
 
-    dataset = IMSDataset(
+    zarr_path = Path(args.ims_zarr)
+    reader = ImzMLReader(
         args.imzml_file, args.ibd_file, micro_res=0.5, ims_res=10
     )
-    dataset.write_zarr(args.ims_zarr, compressor=Zlib(level=1))
-    write_metadata_json(
-        args.ims_metadata,
-        args.zarr_store_url,
-        Path(args.ims_zarr).name,
-        dataset.domain,
-        dataset.transform,
+    reader.to_zarr(str(zarr_path), compressor=Zlib(level=1))
+
+    destination_url = urllib.parse.urljoin(
+        args.server_url, Path(args.ims_zarr).name
+    )
+    write_raster_json(
+        args.raster_json,
+        destination_url,
+        args.raster_name,
+        reader.transform,
+        reader.get_raster_dimensions(),
     )
```

Comment:
Can we move this to a `main` function now that these are getting to be a bit larger?